helpers/sql-obfuscation/test/helpers/sql_obfuscation_test.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+describe OpenTelemetry::Helpers::SqlObfuscation do
+  let(:obfuscation_limit) { 2000 }
+  let(:adapter) { :default }
+  let(:sql) { "SELECT * from users where users.id = 1 and users.email = 'test@test.com'" }
+  let(:expected) { 'SELECT * from users where users.id = ? and users.email = ?' }
+  let(:obfuscate_sql) do
+    OpenTelemetry::Helpers::SqlObfuscation.obfuscate_sql(
+      sql,
+      obfuscation_limit: obfuscation_limit,
+      adapter: adapter
+    )
+  end
+
+  describe '.obfuscate_sql' do
+    it 'returns an obfuscated sql statement' do
+      assert_equal(expected, obfuscate_sql)
+    end
+
+    describe 'when named args with defaults are not passed' do
+      let(:obfuscate_sql) { OpenTelemetry::Helpers::SqlObfuscation.obfuscate_sql(sql) }
+
+      it 'obfuscates the SQL' do
+        assert_equal(expected, obfuscate_sql)
+      end
+    end
+
+    describe 'when sql exceeds obfuscation_limit' do
+      let(:obfuscation_limit) { 42 }
+      let(:expected) { "SELECT * from users where users.id = ...\nSQL truncated (> #{obfuscation_limit} characters)" }
+
+      it 'truncates statements beyond the obfuscation_limit after the first match' do
+        assert_equal(expected, obfuscate_sql)
+      end
+    end
+
+    describe 'when sql has unmatched quote' do
+      let(:sql) { "SELECT * from users where users.id = 1 and users.email = 'test@test.com''" }
+      it 'returns a failure message if unmatched pairs are present' do
+        assert_equal(OpenTelemetry::Helpers::SqlObfuscation::UNMATCHED_PAIRS_FAILURE_MESSAGE, obfuscate_sql)
+      end
+    end
+
+    describe 'when the adapter is mysql' do
+      let(:adapter) { :mysql }
+      let(:sql) { "SELECT * from users where users.id = 1 and users.email = 'test@test.com\255'" }
+
+      it 'encodes utf8' do
```

Comment:
This test seems to indicate the opposite of what I expect. This is what happens when it encounters an error encoding UTF-8 no?

Is there a positive test case example where it successfully encodes SQL with even multi-byte strings like emojis?

Reply:
@arielvalentin  - Thank you so much for noticing this. The UTF-8 encoding was not working at all for the new SQL Obfuscation helper because `opentelemetry-sdk` wasn't a dependency, so `OpenTelemetry::Common` could not be found.

This fix allowed postgres to be correctly encoded with UTF-8 as well! 

See commit: cbf148a623884e9d74d9e885355dae6a84062fee 

Let me know what you think. Once it looks good to you, I'll rebase the commit.